### PR TITLE
Feat/map command

### PR DESCRIPTION
**Pull request by Jakub-Vacek**
## Description

This PR adds `map` command for map preparation

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated the documentation accordingly. For updating Oclif commands documentation use [oclif-dev](https://github.com/oclif/dev-cli#oclif-dev-readme).
- [ ] I have read the **CONTRIBUTION_GUIDE** document.
- [ ] I haven't repeated the code. (DRY)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/commands/map.ts [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    let providerJson: ProviderJson;
+    let profile: { source: string; name: string; scope: string | undefined };
+
+    const tasks = new Listr<{
+      providerName: string | undefined;
+      profileId: string | undefined;
+      quiet: boolean | undefined;
+    }>([
+      {
+        title: 'Loading Comlink interface',
```
Comment (janhalama) [inline src/commands/map.ts:76]: suggestion: I would use `profile` term here, as the command argument `profileId`
```suggestion
        title: 'Loading profile',
```

File: src/commands/map.test.ts [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+          instance.execute({
+            logger,
+            userError,
+            flags: {},
+            args: {
+              providerName: 'test',
+              profileId: `${profileScope}.${profileName}`,
+            },
+          })
+        ).rejects.toThrow(``);
```
Comment (janhalama) [inline src/commands/map.test.ts:166]: question: Is it correct that it throws empty string as error message?

File: src/commands/map.test.ts [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+          .mockResolvedValueOnce(JSON.stringify(providerJson))
+          .mockResolvedValueOnce('something');
+        await expect(
+          instance.execute({
+            logger,
+            userError,
+            flags: {},
+            args: { providerName, profileId: profileName },
+          })
+        ).rejects.toThrow();
```
Comment (janhalama) [inline src/commands/map.test.ts:279]: question: User will not get meaningful error message?

**Review by janhalama — APPROVED** [review]
LGTM, I have some minor questions and text improvements.

File: src/commands/map.test.ts (hunk 2 above)
Comment (Jakub-Vacek) [reply src/commands/map.test.ts]: Nope, I will add expected errors

File: src/commands/map.test.ts (hunk 3 above)
Comment (Jakub-Vacek) [reply src/commands/map.test.ts]: He will get meaningfull message:)

**Pull request merged**